BE: Profile Model

dproc96 opened the pull request:
**What it does**
- Create a DB model for the Profile
- Create test cases for the Profile model
- Create file for future test cases for the Database
- Create index.js for grouping models (duplicated work in PR #24)

**What to look for**
- Any validators I should add to the Model
- Any statics or methods that would be useful to add
- Any test cases I should add

**What to avoid**
- I used my instincts for what should be required, if more data should be required I'll add validators for that too

dproc96 committed:
Create Profile.js

dproc96 committed:
Create index.js

To group together the Models (Michael B did something similar in his, so we can just combine them later)

dproc96 committed:
Create Database.spec.js

dproc96 committed:
Update Database.spec.js

Added first test cases for successful doc save

dproc96 committed:
Update Database.spec.js

Added enum validation check and closed connection at test completion

dproc96 committed:
Update Database.spec.js

Added error tests

dproc96 committed:
Update Database.spec.js

Added second error test

dproc96 committed:
Update Database.spec.js

Added retrieval test

Braagaa commented:
Maybe we can add an age restriction validator to the birthDay attribute. Or to check and make sure that the availability end date is not less than the beginning availability date. Anyone else agree?

dproc96 commented:
> Maybe we can add an age restriction validator to the birthDay attribute. Or to check and make sure that the availability end date is not less than the beginning availability date. Anyone else agree?

Oh those are good, I'll add those in! Thanks Michael!

dproc96 committed:
Update Profile.js

Add validators

dproc96 committed:
Update Database.spec.js

Added test for under 18

dproc96 committed:
Update Profile.js

Fixed endless save bug

dproc96 committed:
Update Profile.js

Fixed validators where they were crashing the app

dproc96 committed:
Update Database.spec.js

Added test case for negative availability range

dproc96 commented:
@Braagaa I've added your suggestions to the code and they're working!

Braagaa commented:
:+1:

Braagaa commented:
I think we missing some attributes. The mocks are showing email, address, and phone number. We possibly need a profile image url as well.

dahrk commented:
Michael B made a good point, there are some more fields that can be added.
A style tip would be to add newlines at the end of your files

dahrk commented on server/test/Database.spec.js:
```diff
@@ -0,0 +1,130 @@
+        })
+      })
+
+      it("Rejects user if under 18", (done) => {
```
Something to verify - if no date is entered, will the user be denied? If this check is being run, should birthdate be required as well?

aeciorc reviewed (commented):
Looks good Dan. One thing we need to discuss better is the distinction between a regular user and a sitter. For a regular user, for example, the availability wouldn't be required. We can talk about that during our standup

dproc96 commented:
@Braagaa @dahrk Yeah, those would be good things to include, I was following the ticket as written but we can discuss modifications

dproc96 committed:
Update Profile.js

Added isSitter and made birthdate required (removed unecessary conditional now that birthdate is required)

dproc96 committed:
Update Database.spec.js

Altered tests and added one to accomodate required birth date

dproc96 committed:
Update Database.spec.js

Add tests for new field isSitter

dproc96 committed:
Update Profile.js

Added phone number and location as fields

dproc96 committed:
Update Database.spec.js

Added test cases for the new fields in Profile model
